src/main/java/spoon/MavenLauncher.java

```diff
@@ -108,656 +125,144 @@ public MavenLauncher(String mavenProject, String m2RepositoryPath, SOURCE_TYPE s
-				return null;
-			}
-			TreeDependency dependence = new TreeDependency(groupId, artifactId, version, dependency.getType());
-			try {
-				InheritanceModel dependencyModel = readPom(groupId, artifactId, version);
-				if (dependencyModel != null) {
-					dependence = dependencyModel.getTreeDependency(true, hierarchy);
-					dependence.groupId = groupId;
-					dependence.artifactId = artifactId;
-					dependence.version = version;
+	static String guessMavenHome() {
+		String mvnHome = null;
+		try {
+			String[] cmd = {"mvn", "-version"};
+			Process p = Runtime.getRuntime().exec(cmd);
+			BufferedReader output = new BufferedReader(new InputStreamReader(p.getInputStream()));
+			String line;
 
-					if (dependency.getExclusions() != null) {
-						for (int i = 0; i < dependency.getExclusions().size(); i++) {
-							Exclusion exclusion = dependency.getExclusions().get(i);
-							dependence.removeDependency(exclusion.getGroupId(), exclusion.getArtifactId());
-						}
-					}
+			while ((line = output.readLine()) != null) {
+				if (line.contains("Maven home: ")) {
+					mvnHome = line.replace("Maven home: ", "");
+					return mvnHome;
 				}
-
-			} catch (Exception e) {
-				LOGGER.log(Level.ERROR, "Unable to read the pom of the dependency:" + dependence.toString(), e);
 			}
-			return dependence;
-		}
 
-		private TreeDependency getTreeDependency() {
-			return getTreeDependency(false, new HashSet<>());
-		}
-
-		/**
-		 * Get the list of dependencies available in the local maven repository
-		 *
-		 * @param isLib: If false take dependency of the main project; if true, take dependencies of a library of the project
-		 * @return the list of  dependencies
-		 */
-		private TreeDependency getTreeDependency(boolean isLib, Set<TreeDependency> hierarchy) {
-			String groupId = extractVariable(model.getGroupId());
-			String artifactId = extractVariable(model.getArtifactId());
-			String version = extractVersion(groupId, artifactId, model.getVersion());
-			TreeDependency dependence = new TreeDependency(groupId, artifactId, version, model.getPackaging());
-			if (hierarchy.contains(dependence)) {
-				return dependence;
-			}
-			hierarchy.add(dependence);
-
-
-			// add the parent has a dependency
-			if (this.parent != null) {
-				dependence.addDependence(this.parent.getTreeDependency(isLib, hierarchy));
-			}
-
-			List<Dependency> dependencies = model.getDependencies();
-			for (Dependency dependency : dependencies) {
-				dependence.addDependence(getTreeDependency(dependency, isLib, hierarchy));
-			}
-
-			if (!isLib) {
-				for (InheritanceModel module : modules) {
-					if (module.model.getGroupId() == null) {
-						module.model.setGroupId(groupId);
-					}
-					if (module.model.getVersion() == null) {
-						module.model.setVersion(version);
-					}
-					dependence.addDependence(module.getTreeDependency(isLib, hierarchy));
-				}
-			}
-			return dependence;
+			p.waitFor();
+		} catch (IOException e) {
+			e.printStackTrace();
```

Comment:
Here again: declaring two exception for the same behaviour is useless. You can catch both in the same block and throw a SpoonException here.

Reply:
Ok.